jdrfandroidbleparser/src/test/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/characteristic/compoundcharacteristic/GlucoseMeasurementCharacteristicTest.kt

```diff
@@ -0,0 +1,202 @@
+        assertEquals(true, glucoseMeasureCharacteristic.flags?.contains(Flags.CONCENTRATION_PRESENT))
+        assertEquals(false, glucoseMeasureCharacteristic.flags?.contains(Flags.CONTEXT_PRESENT))
+        assertEquals(false, glucoseMeasureCharacteristic.flags?.contains(Flags.STATUS_ANNUNCIATION_PRESENT))
+        assertEquals(false, glucoseMeasureCharacteristic.flags?.contains(Flags.CONCENTRATION_UNIT))
+        //verify the sequence number part
+        assertEquals(4, glucoseMeasureCharacteristic.sequenceNumber)
+        //verify the base time
+        assertEquals(2018, glucoseMeasureCharacteristic.timeYear)
+        assertEquals(6, glucoseMeasureCharacteristic.timeMonth)
+        assertEquals(6, glucoseMeasureCharacteristic.timeDay)
+        assertEquals(14, glucoseMeasureCharacteristic.timeHour)
+        assertEquals(57, glucoseMeasureCharacteristic.timeMinute)
+        assertEquals(4, glucoseMeasureCharacteristic.timeSecond)
+        //verify the offset part value
+        assertEquals(0, glucoseMeasureCharacteristic.timeOffset)
+        //verify the glucose measurement value part
+        assertEquals(true, (glucoseMeasureCharacteristic.concentration!! - 0.00117f) < ERROR_ACCEPTABLE)
+        //verify the sample location field
+        assertEquals(SampleLocation.FINGER, glucoseMeasureCharacteristic.sampleLocation)
+        //verify the sample type field
+        assertEquals(Type.WHOLE_BLOOD_CAPILLARY, glucoseMeasureCharacteristic.fluidType)
+        //verify the unit
+        assertEquals(Units.KILOGRAM_PER_LITRE, glucoseMeasureCharacteristic.unit)
+
+        //test the user display time is correct (time + offset)
+        val expectedDate = Date(118, 5, 6, 14, 57, 4)
+        assertEquals(expectedDate.time, glucoseMeasureCharacteristic.displayTime?.time)
+    }
+
+    //The following test the univer conversion accuracy
+    @Test
+    fun getConcentration() {
+        val testRawPacket = byteArrayOf(0x03, 0x04, 0x00, 0xe2.toByte(), 0x07, 0x06, 0x06, 0x0e, 0x39, 0x04, 0x00, 0x00, 0x3f, 0xb0.toByte(), 0x11.toByte())
+
+        val characteristics = mockBTCharacteristic(testRawPacket)
+        val glucoseMeasureCharacteristic = GlucoseMeasurementCharacteristic(characteristics)
+        //verify the glucose measurement value part
+        assertEquals(true, (glucoseMeasureCharacteristic.concentration!! - 0.00117f) < ERROR_ACCEPTABLE)
+        //converted 0.00117f to 117 mg/dL
+        assertEquals(true, (glucoseMeasureCharacteristic.getConcentration(Units.MILLIGRAM_PER_DECILITRE)!! - 117f) < ERROR_ACCEPTABLE)
+
+        //convert 117f mg/dL to 6.5 mmol/L
+        assertEquals(true, (glucoseMeasureCharacteristic.getConcentration(Units.MILLIMOLE_PER_LITRE)!! - 6.5f) < ERROR_ACCEPTABLE)
+
+        //convert 117f mg/dL to 0.0065mol/L
+        assertEquals(true, (glucoseMeasureCharacteristic.getConcentration(Units.MOLE_PER_LITRE)!! - 0.0065f) < ERROR_ACCEPTABLE)
+
+    }
+
+
+}
```

Comment:
Take a look at the tests I wrote for the Glucose Context, and model your tests to look like those. These are hard to read and follow.